### PR TITLE
introduce feature list

### PR DESCRIPTION
**Pull request by LHT129**
#154

### CODE REVIEW

File: src/index_feature_list_test.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+TEST_CASE("Test set&check", "[ut][index_feature_list]") {
+    using namespace vsag;
+    IndexFeatureList list;
+    auto count = static_cast<uint32_t>(IndexFeature::INDEX_FEATURE_COUNT);
+    for (uint32_t i = 0; i < count; ++i) {
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == false);
+        list.SetFeature(static_cast<IndexFeature>(i));
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == true);
+        list.SetFeature(static_cast<IndexFeature>(i), false);
+        REQUIRE(list.CheckFeature(static_cast<IndexFeature>(i)) == false);
```
Comment (inabao) [inline src/index_feature_list_test.cpp:29]: We need to verify that the current feature's settings do not affect other features.

Comment (LHT129) [reply src/index_feature_list_test.cpp:29]: done

**Review by inabao — APPROVED** [review]
LGTM

File: include/vsag/index_feature.h [hunk 2]
```diff
@@ -17,8 +17,8 @@
 
 namespace vsag {
 
-enum class IndexFeature {
-    NEED_TRAIN, /**< Indicates that the class needs to be trained */
+enum IndexFeature {
+    NEED_TRAIN = 0, /**< Indicates that the class needs to be trained */
```
Comment (wxyucs) [inline include/vsag/index_feature.h]: start from 1, 0 is reserved

**Review by wxyucs — CHANGES_REQUESTED** [review]
@LHT129 please add an example and an issue of this feature

File: include/vsag/index_feature.h (hunk 2 above)
Comment (LHT129) [reply include/vsag/index_feature.h]: why reserve 0, I have make the last one is the count of the enum, if start with 1,  when I cast 0 to a IndexFeature type will cause error.

Comment (wxyucs) [reply include/vsag/index_feature.h]: reserve of 0 is useful for exposing the problems with uninitiated variables or unexpected type casting

Comment (LHT129) [reply include/vsag/index_feature.h]: done

**Pull request merged**